gitlab-status/handler_test.go

```diff
@@ -73,47 +76,264 @@ func Test_appendParameters(t *testing.T) {
+			url:         "https://some.random.url/api/v4/projects/3/statuses/99a7c6009?value=somevalue&target_url=https://www.world.com/",
 			state:       "failure",
 			desc:        "some description",
 			context:     "context",
-			expectedURL: "https://some.random.url/api/v4/projects/3/statuses/99a7c6009?context=context&description=some+description&state=failed",
+			targetURL:   "https://www.world.com/",
+			expectedURL: "https://some.random.url/api/v4/projects/3/statuses/99a7c6009?context=context&description=some+description&state=failed&target_url=https%3A%2F%2Fwww.world.com%2F",
 		},
 		{
 			title:       "When state is empty",
 			url:         "https://some.random.url/api/v4/projects/3/statuses/99a7c6009?value=somevalue",
 			state:       "",
 			desc:        "some description",
 			context:     "stack-deploy",
-			expectedURL: "https://some.random.url/api/v4/projects/3/statuses/99a7c6009?context=stack-deploy&description=some+description&state=",
+			targetURL:   "https://www.world.com/",
+			expectedURL: "https://some.random.url/api/v4/projects/3/statuses/99a7c6009?context=stack-deploy&description=some+description&state=&target_url=https%3A%2F%2Fwww.world.com%2F",
 		},
 		{
 			title:       "When everything is empty",
 			url:         "https://some.random.url/api/v4/projects/3/statuses/99a7c6009?value=somevalue",
 			state:       "",
 			desc:        "",
 			context:     "",
-			expectedURL: "https://some.random.url/api/v4/projects/3/statuses/99a7c6009?context=&description=&state=",
+			targetURL:   "",
+			expectedURL: "https://some.random.url/api/v4/projects/3/statuses/99a7c6009?context=&description=&state=&target_url=",
 		},
 	}
 	for _, test := range tests {
 		t.Run(test.title, func(t *testing.T) {
-			wholeUrl, _ := appendParameters(test.url, test.state, test.desc, test.context)
+			wholeUrl, _ := appendParameters(test.url, test.state, test.desc, test.context, test.targetURL)
 			if wholeUrl != test.expectedURL {
 				t.Errorf("wanted: %s got: %s", test.expectedURL, wholeUrl)
 			}
 		})
 	}
 }
+
+func Test_FormatLogsURL(t *testing.T) {
+	tests := []struct {
+		title       string
+		event       *sdk.Event
+		gatewayURL  string
+		expectedURL string
+	}{
+		{
+			title: "gateway_public_url is set and object is set",
+			event: &sdk.Event{
+				Owner:      "exampleusername",
+				Service:    "exampleservice",
+				Repository: "examplerepository",
+				SHA:        "98zxc823axcc",
+			},
+			gatewayURL:  "https://cloud.o6s.io/",
+			expectedURL: "https://system.o6s.io/dashboard/exampleusername/exampleservice/log?repoPath=exampleusername/examplerepository&commitSHA=98zxc823axcc",
+		},
+		{
+			title:       "gateway_public_url is set but the object is not populated",
+			event:       &sdk.Event{},
+			gatewayURL:  "https://cloud.o6s.io/",
+			expectedURL: "https://system.o6s.io/dashboard///log?repoPath=/&commitSHA=",
+		},
+		{
+			title: "gateway_public_url is not set right but the object is populated right",
```

Comment:
"not set right" - what does that mean when to a new contributor reading the code in 6 months? Can we improve the text?

Reply:
Yeah you are right, I didn't think about this in that way, I will fix the titles

Reply:
Done, added better explanation